### ISSUE TITLE
Can't retry login code entry

### ISSUE THREAD

**Issue by abiro** [aws-samples/amazon-cognito-passwordless-email-auth#17]
The back end should allow 3 attempts for entering the login code, but the session is invalidated in the client after the first invalid attempt.

### Steps to reproduce:

1. Request login code
2. Submit invalid login code
3. Submit correct login code

### Expected result:

The user should be logged in.

### Actual result

The user is not logged in.

### Additional context

The first `Auth.sendCustomChallengeAnswer` request with the INVALID code returns the following error:

`Object { code: "UserLambdaValidationException", name: "UserLambdaValidationException", message: "CreateAuthChallenge failed with error 'challengeMetaData'." }`

The second `Auth.sendCustomChallengeAnswer` request with the CORRECT code returns the following error:

`Object { code: "NotAuthorizedException", name: "NotAuthorizedException", message: "Invalid session for the user." }`

**Comment by abiro** [comment]
Apologies, the bug is not reproducible with this repo. 

I traced the error back to my code and ultimately the documentation. I've implemented handlers following this repo in typed Python. I've created type definitions for the trigger events based on the docs and it seems there is an error in the docs for [Create Auth Challenge Lambda Trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-create-auth-challenge.html):

ChallengeResult.challengeMeta**D**ata should be ChallengeResult.challengeMeta**d**ata

I've submitted an error feedback.
